### ISSUE TITLE
Incorrect nullness annotations for java.lang.ThreadLocal

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#125]
```
The following code contains a nullness error:

    public class A {
        public static void main( String[] args ) {
            ThreadLocal<Object> tl = new ThreadLocal<Object>( );

            // Always throws an NPE
            tl.get( ).toString( );
        }
    }

However, the error is not caught by the nullness checker because the annotated 
JDK specifies a default of @NonNull for the class. This affects 
java.lang.InheritableThreadLocal as well.
```

Original issue reported on code.google.com by `atomknig...@gmail.com` on 5 Feb 2012 at 3:00
